soccer/gameplay/tactics/positions/submissive_defender.py

```diff
@@ -29,10 +34,45 @@ def __init__(self):
 
         self.add_state(SubmissiveDefender.State.marking,
                        behavior.Behavior.State.running)
+        self.add_state(SubmissiveDefender.State.clearing,
+                       behavior.Behavior.State.running)
 
         self.add_transition(behavior.Behavior.State.start,
                             SubmissiveDefender.State.marking, lambda: True,
                             "immediately")
+        self.add_transition(SubmissiveDefender.State.marking,
+                            SubmissiveDefender.State.clearing,
+                            lambda: self.go_clear,
+                            "when it is safe to clear the ball")
+        self.add_transition(
+            SubmissiveDefender.State.clearing,
+            SubmissiveDefender.State.marking,
+            lambda: self.subbehavior_with_name('kick-clear').state == behavior.Behavior.State.completed or not self.should_clear_ball(evaluation.ball.time_to_ball(self.robot)),
+            "done clearing")
+
+    def should_clear_ball(self, our_time_to_ball):
+        #Returns true if our robot can reach the ball sooner than the closest opponent
+        if main.ball().pos.mag() < constants.Field.ArcRadius * 2:
+            safe_to_clear = True
+            #change this to use the config system when we figure out how to do that
+            max_vel = 3.5
+            max_accel = 1.8
+
+            for robot in main.system_state().their_robots:
+                their_dist_to_ball = robot.pos.dist_to(main.ball().pos)
+                #if their robot is moving faster than ours, assume it is at its maximum speed, otherwise assume its max speed is the same as ours
+                their_max_vel = max(max_vel, robot.vel.mag())
```

Comment:
It may be best to just always assume their robots can go as fast as ours.  Often times our defenders are stationary, so with this code, an enemy that's barely moving will be assumed to be at its max speed.